webpack/assets/javascripts/react_app/redux/actions/notifications/index.js

```diff
@@ -134,7 +135,22 @@ export const toggleDrawer = () => (dispatch, getState) => {
   });
 };
 
-export const clickedLink = link => (dispatch, getState) => {
-  toggleDrawer()(dispatch, getState);
-  window.open(link.href, link.external ? '_blank' : '_self');
+export const clickedLink = (
+  { href, external = false },
+  toggleDrawerAction = toggleDrawer,
+) => (dispatch) => {
+  dispatch(toggleDrawerAction());
+
+  const openedWindow = window.open(href, external ? '_blank' : '_self');
+
+  if (external) {
+    openedWindow.opener = null;
+  }
+
+  dispatch({
+    type: NOTIFICATIONS_LINK_CLICKED,
```

Comment:
what am I missing, where is the reducer part?

Reply:
There is no reducer for link-clicked (add there is no need to) but I still see the value of dispatching the message.

1. This is what actions should do
2. It helps with testing
3. It helps with debugging
4. For future use